src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+  color: 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Figure['color'];
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Figure['color'], a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides of a triangle must be positive numbers');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Parametrs do not form a triangle');
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
```

Comment:
The `shape` property should be declared as a type literal `'circle'` rather than a string literal type. This is correct as per TypeScript's handling of literal types.